src/AstGen.zig

```diff
@@ -1789,7 +1790,6 @@ fn structInitExpr(
         .none => return structInitExprAnon(gz, scope, node, struct_init),
         .discard => {
             // Even if discarding we must perform an anonymous init to check for duplicate field names.
-            // TODO: should duplicate field names be caught in AstGen?
```

Comment:
The deletion of this comment indicates that this logic is no longer necessary. Either rework the logic (to evaluate each field individually with the `discard` result location rather than constructing any aggregate), or put a new TODO comment here.

Reply:
I don't want to put another TODO :smile:
Please explain a little more exactly what you want me to do.

As I currently understand, this is no longer needed?

Reply:
This corresponds to code like `_ = .{ .a = 0, .b = 1 }`. Rather than actually initing a struct temporary, this should just evaluate `0` and `1` with `ResultInfo.Loc.discard`, as if you wrote `_ = 0; _ = 1;`.

Reply:
Are you saying that this type of discarded struct should not be checked for duplicates?

Reply:
No, that is absolutely not what I am saying. As the removed comment mentions, the current logic - actually constructing a struct temporary - exists _in order to_ ensure this error is triggered. Since the logic is now moved to AstGen, it is not necessarily to construct a temporary anymore.

Reply:
ok, thanks for clarifying. that's what i thought in my first message, but i began to doubt my understanding of english :smile:

i've looked through the code, and thought about what you said, and i understand now!

will just this be fine: `.discard => return .void_value,`

Reply:
No - as I have said, it is still important to evaluate the sub-expressions, for any compilation errors or side effects. Before returning `.void_value`, call `expr` on each initialization sub-expression with `.{ .rl = .discard }` as the `ri`.